src/NDItk/multigroup/OutgoingParticleTypes/test/OutgoingParticleTypes.test.cpp

```diff
@@ -0,0 +1,121 @@
+// include Catch2
+#include <catch2/catch_test_macros.hpp>
+#include <catch2/matchers/catch_matchers_floating_point.hpp>
+using Catch::Matchers::WithinRel;
+
+// what we are testing
+#include "NDItk/multigroup/OutgoingParticleTypes.hpp"
+
+// other includes
+
+// convenience typedefs
+using namespace njoy::NDItk;
+using OutgoingParticleTypes = multigroup::OutgoingParticleTypes;
+
+std::string chunk();
+void verifyChunk( const OutgoingParticleTypes& );
+std::string chunkWithInsufficientNumberValues();
+
+SCENARIO( "OutgoingParticleTypes" ) {
+
+  GIVEN( "valid data for an OutgoingParticleTypes instance" ) {
+
+    std::string record = chunk();
+
+    WHEN( "the data is given explicitly" ) {
+
+      std::vector< int > values = { 0, 1001 };
+
+      OutgoingParticleTypes chunk( std::move( values ) );
+
+      THEN( "an OutgoingParticleTypes can be constructed and members can "
+            "be tested" ) {
+
+        verifyChunk( chunk );
+      } // THEN
+
+      THEN( "the record can be printed" ) {
+
+        std::string buffer;
+        auto output = std::back_inserter( buffer );
+        chunk.print( output );
+
+        CHECK( buffer == record );
+      } // THEN
+    } // WHEN
+
+    WHEN( "the data is read using iterators" ) {
+
+      auto iter = record.begin() + 14;
+      auto end = record.end();
+
+      OutgoingParticleTypes chunk;
+      chunk.read( iter, end, 2 );
+
+      THEN( "an OutgoingParticleTypes can be constructed and members can "
+            "be tested" ) {
+
+        verifyChunk( chunk );
+      } // THEN
+
+      THEN( "the record can be printed" ) {
+
+        std::string buffer;
+        auto output = std::back_inserter( buffer );
+        chunk.print( output );
+
+        CHECK( buffer == record );
+      } // THEN
+    } // WHEN
+  } // GIVEN
+
+  GIVEN( "invalid data for an OutgoingParticleTypes instance" ) {
+
+    WHEN( "the number of weight values is insufficient" ) {
```

Comment:
OutgoingParticleTypes, not weight values.

Reply:
copy/paste again